rflx/model/message.py

```diff
@@ -640,6 +641,62 @@ def __compute_field_condition(self, final: Field) -> expr.Expr:
         )
 
 
+class ParallelProofs:
+    def __init__(self) -> None:
+        self.proofs: List[
+            List[Tuple[expr.Expr, List[expr.Expr], expr.ProofResult, RecordFluxError, bool, bool]]
+        ] = []
+        self.current: List[
+            Tuple[expr.Expr, List[expr.Expr], expr.ProofResult, RecordFluxError, bool, bool]
+        ] = []
+
+    def add(
+        self,
+        goal: expr.Expr,
+        facts: List[expr.Expr],
+        expected: expr.ProofResult,
+        message: RecordFluxError,
+        negate: bool = False,
+        add_unsat: bool = False,
+    ) -> None:
+        # pylint: disable=too-many-arguments
+        self.current.append((goal, facts, expected, message, negate, add_unsat))
+
+    def push(self) -> None:
+        if self.current:
+            self.proofs.append(copy(self.current))
+            self.current.clear()
+
+    @classmethod
+    def check_proof(
+        cls,
+        argument: List[
+            Tuple[expr.Expr, List[expr.Expr], expr.ProofResult, RecordFluxError, bool, bool]
+        ],
```

Comment:
IMHO, `argument` is not a meaningful name.

Reply:
Changed in 9fe084ffc5266477649107f038682ca9997ed9a3.